File: Solution Using C#/Fawry.Ecommerce/ShipDecorator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Fawry.Ecommerce
{
    internal class ShipDecorator:Product,IShippable
    {
        private readonly Product _product;
        private readonly double _weight;
        public ShipDecorator(Product product,double weight) : base(product.Name, product.Price, product.Quantity)
        {
            _product = product;
            _weight = weight;
        }
        public override bool IsExpired()
        {
            return _product.IsExpired();
        }
        public string GetName()
        {
            return _product.Name;
        }

        public double GetWeight()
        {
            return  _weight;
        }
    }
}

File: Solution Using Java/src/models/ShipDecorator.java
package models;

public class ShipDecorator extends Product implements IShippable{
    private final Product product;
    private final double weight;
    public ShipDecorator(Product product,double weight) {
        super(product.getName(),product.getPrice(),product.getQuantity());
        this.product = product;
        this.weight = weight;
    }

    @Override
    public boolean isExpired() {
        return product.isExpired();
    }
    @Override
    public String getName(){
        return product.getName();
    }
    @Override
    public double getWeight() {
        return weight;
    }
}
